features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Listing/SampleAPI/SampleAPI.jsx

```diff
@@ -98,21 +98,33 @@ class SampleAPI extends Component {
             console.error(error);
             Alert.error(error);
         });
-        swaggerUpdatePromise.then((sampleAPI) => {
-            sampleAPI.publish()
-                .then(() => {
-                    this.setState({ published: true, api: sampleAPI });
-                    Alert.info(intl.formatMessage({
-                        id: 'Apis.Listing.SampleAPI.SampleAPI.created',
-                        defaultMessage: 'Sample PizzaShackAPI API created successfully',
-                    }));
-                })
+        if (!AuthManager.isNotPublisher()) {
+            swaggerUpdatePromise.then((sampleAPI) => {
+                sampleAPI.publish()
+                    .then(() => {
+                        this.setState({ published: true, api: sampleAPI });
+                        Alert.info(intl.formatMessage({
+                            id: 'Apis.Listing.SampleAPI.SampleAPI.created',
+                            defaultMessage: 'Sample PizzaShackAPI API created successfully',
+                        }));
+                    })
+                    .catch((error) => {
+                        this.setState({ deploying: false });
+                        Alert.error(error);
+                    });
+            });
+        } else {
+            swaggerUpdatePromise.then((sampleApi) => {
+                this.setState({ published: true, api: sampleApi });
+                Alert.info(intl.formatMessage({
+                    id: 'Apis.Listing.SampleAPI.SampleAPI.created',
+                    defaultMessage: 'Sample PizzaShackAPI API created successfully',
+                }));
+            })
                 .catch((error) => {
-                    console.error(error);
-                    this.setState({ deploying: false });
```

Comment:
catch doesn't seem to be indented properly

Reply:
The indentation level is correct. Changing it would throw node build errors.